spec/models/framework/definition/ast/field_spec.rb

```diff
@@ -45,22 +46,42 @@
     end
 
     context 'with a lookup (non-primitive) type' do
-      let(:field_def) do
-        {
-          kind: :additional,
-          optional: true,
-          type:  'PaymentProfile',
-          field: 'Additional2',
-          from:  'Payment Profile'
-        }
-      end
+      context 'on an Additional field' do
```

Comment:
So there is some inherent knowledge that is required to be able to make sense of these nested context: that a "type" that isn't a primitive type, or a "known" type _must_ be a lookup and that the presence of a corresponding `Lookups` in the definition is assumed. Is there some way that this can be made clearer to someone who might not be so close to FDL and its development? I'm wondering if being explicit and passing in a hash containing the lookup for `PaymentProfile` would help avoid confusion (even if the underlying code doesn't actually need its presence to have this test pass).

Reply:
Yes, these tests are useful but not very readable. All of the `field_def` hashes are hashes that would have come from an AST. My suggestion is that we move to an integration test model here and put the real FDL in the tests and *then* say how that FDL would relate to a `Field`. That would also reduce the scope for these AST fragments falling out of sync with the parser. We *should* also be able to make the `context` and `describe` blocks do a lot more work to reflect what's really happening.

Reply:
I wouldn't want to pass a hash we don't use; this would be if not linted out of existence now, then terminated with extreme prejudice by a future developer lacking this context (hey, we're *ruthless*, we are)

Reply:
Resolved in https://github.com/dxw/DataSubmissionServiceAPI/pull/318/commits/4925b8e723321f5515ec67f0620e36df51840e23